### PR TITLE
Add RNN operator

### LINKED ISSUE #78: Implement RNN operator
https://github.com/onnx/onnx/blob/main/docs/Changelog.md#RNN-7

### PR DESCRIPTION
**Pull request by Swopper050**
Closes #78

### CODE REVIEW

File: ops/opset13/rnn.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+	seqLength := X.Shape()[0]
+	batchSize := X.Shape()[1]
+
+	Ht := inputs[5]
+	if Ht == nil {
+		Ht = r.getInitialH(batchSize)
+	}
+
+	// Reshape the hidden tensor without the bidirectional dimension, as
+	// we do not support bidirectional RNN yet. This is the first dimension.
```
Comment (wipsel) [inline ops/opset13/rnn.go:115]: "This is the first dimension"?

File: ops/opset13/rnn.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+			return nil, err
+		}
+	}
+
+	Yh, ok := Ht.Clone().(tensor.Tensor)
+	if !ok {
+		return nil, ops.ErrTypeAssert("tensor.Tensor", Ht.Clone())
+	}
+
+	// Reshape the outputs so it adds the num_directions as specified by onnx.
```
Comment (wipsel) [inline ops/opset13/rnn.go:156]: You mention `num_directions` in the comment but it is not clear to me where you use this?

File: ops/opset13/opset13_test.go [hunk 3]
```diff
@@ -33,16 +33,16 @@ func TestGetOperator(t *testing.T) {
...
-      newAtan(),
-      nil,
-    },
-    {
-      "Atanh",
-      newAtanh(),
-      nil,
-    },
+		{
+			"Atan",
```
Comment (wipsel) [inline ops/opset13/opset13_test.go:37]: Are these changes belonging here?

Comment (Swopper050) [reply ops/opset13/opset13_test.go:37]: somehow they were not formatted right, so I'd say it is nice to format them right and we can include it here.

Comment (Swopper050) [reply ops/opset13/opset13_test.go]: However, the formatting was already fixed in another merge request, so the changes are not needed anymore.

File: ops/opset13/relu.go [hunk 4]
```diff
@@ -21,24 +21,9 @@ func (r *Relu) Init(_ []*onnx.AttributeProto) error {
 
 // Apply applies the relu operator.
 func (r *Relu) Apply(inputs []tensor.Tensor) ([]tensor.Tensor, error) {
-	t := inputs[0]
+	out, err := ops.ReLU(inputs[0])
```
Comment (wipsel) [inline ops/opset13/relu.go:24]: maybe check the err and return nil, instead of []tensor.Tensor{nil} if an error occurs?

File: ops/opset13/rnn.go [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+type RNNDirection string
+
+const (
+	Forward       RNNDirection = "forward"
+	Reverse       RNNDirection = "reverse"
+	Bidirectional RNNDirection = "bidirectional"
+)
+
+// These activations are supported in the RNN calculation.
+var RNNActivations = map[string]ops.Activation{
```
Comment (wipsel) [inline ops/opset13/rnn.go:26]: See comment in gru PR

File: ops/opset13/rnn.go [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+	MaxRNNInputs = 6
+)
+
+// RNNDirection is the direction of the RNN. RNNs process sequences. We can process
+// those forward (from first to last), in reverse (from last to first) or
+// bidirectional (which is both forward and reverse added together).
+type RNNDirection string
+
+const (
+	Forward       RNNDirection = "forward"
```
Comment (wipsel) [inline ops/opset13/rnn.go:20]: Do these need to be exported?

**Review by wipsel — DISMISSED** [review]

**Pull request merged**